File: 1271_hexspeak/solution.py
class Solution:
    def toHexspeak(self, num: str) -> str:
        value = int(num)
        digits = "0123456789ABCDEF"
        out = ""
        while value:
            value, rem = divmod(value, 16)
            if 2 <= rem <= 9:
                return "ERROR"
            out = digits[rem] + out
        return (out or "0").replace("0", "O").replace("1", "I")

File: 1271_hexspeak/Solution.java
// LeetCode 1271 - Hexspeak
// https://leetcode.com/problems/hexspeak/

import java.math.BigInteger;

class Solution {
    public String toHexspeak(String num) {
        BigInteger value = new BigInteger(num);
        String digits = "0123456789ABCDEF";
        StringBuilder out = new StringBuilder();
        while (value.signum() > 0) {
            int rem = value.mod(BigInteger.valueOf(16)).intValue();
            if (rem >= 2 && rem <= 9) return "ERROR";
            out.insert(0, digits.charAt(rem));
            value = value.divide(BigInteger.valueOf(16));
        }
        String result = out.length() == 0 ? "0" : out.toString();
        return result.replace('0', 'O').replace('1', 'I');
    }
}
